extensions/amp-script/0.1/amp-script.js

```diff
@@ -111,13 +112,37 @@ export class AmpScript extends AMP.BaseElement {
 
   /** @override */
   buildCallback() {
-    this.development_ = this.element.hasAttribute('development');
+    /*
+     * Development mode is enabled by satisfying two constraints:
+     *   1. Root html element has 'data-ampdevmode'
+     *   2. The amp-script tag or any of its parents must also have 'data-ampdevmode'.
+     */
+    const htmlEl = this.element.ownerDocument.documentElement;
+    let devMode = false;
+    if (htmlEl.hasAttribute('data-ampdevmode')) {
+      const closestAncestorWithDevTag = closestAncestorElementBySelector(
+        this.element,
+        'data-ampdevmode'
+      );
+      devMode =
+        closestAncestorWithDevTag != null &&
```

Comment:
It can never be null, since we know the `<html>` has it. 😉

Reply:
good realization! lets me get rid of the if statement.